Create CONTRIBUTING.md (#142)
mlx-graphs/mlx-graphs: CONTRIBUTING.md

# Contributing guidelines

## Opening an Issue

- Before [creating an issue](https://help.github.com/en/github/managing-your-work-on-github/creating-an-issue), check if you are using the latest version of the project. If you are not up-to-date, see if updating fixes your issue first.
- Do not to open duplicate issues**; before creating a new issue, go through existing ones to see if it has previously been reported. If your issue exists, comment with any additional information you have.

> - Select the appropriate issue template and fully complete it. If no template is available for your specific issue, please use a blank one and report as much details as needed

Reviewer: ```suggestion
- Select the appropriate issue template and fully complete it. If no template is available for your specific issue, please use a blank one and report as much details as needed.
```

### Reporting bugs

A great way to contribute to the project is to open a detailed (bug) issue when you encounter a problem. We use a custom template for bugs. Make sure you're using the latest version of the project before opening a bug issue.

### Feature Requests

Another great way to contribute is to open feature requests!
While all requests are valid, we cannot guarantee your request will be accepted. Your idea may be great, but also out-of-scope for the project.
For every feature request, you are welcome to submit a pull request to help!

## Asking Questions
Please use [Discussions](https://github.com/mlx-graphs/mlx-graphs/discussions) to ask questions instead of issues.
GitHub issues are not the appropriate place to debug your specific project, but should be reserved for filing bugs and feature requests.

> 

Reviewer: ```suggestion
To chat and share random questions with the community, we recommend to join our [Discord server](https://github.com/mlx-graphs/mlx-graphs/discussions/107).

```

Author: added the actual link

## Pull Requests
When you submit a pull request, you'll be presented with a template to fill with a description of your contribution and checks on whether you added tests and documentation. Please fill all the parts.
To open a PR:

> - Fork and submit pull requests to the repo.

Reviewer: ```suggestion
- Fork the repo and follow the [local installation guidelines](https://github.com/mlx-graphs/mlx-graphs?tab=readme-ov-file#contributing).
- Add your contribution and submit a pull request to the repo.
```

- If you've added code that should be tested, add tests.

> - Update the necessary documentation

Reviewer: ```suggestion
- Update the necessary documentation.
```
